src/root/types/project.ts

```diff
@@ -0,0 +1,93 @@
+import {
```

Comment:
Let's not do this.
We should try to write types where it is used.

Reply:
It is used in a lot of place and we might have cyclic dependency if not separated out

Reply:
If it's used everywhere then it's okay.

P.S. Importing types does not create cyclic dependencies.